projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -34,32 +34,56 @@
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using System.Text;
+
 using RabbitMQ.Client.Exceptions;
 using RabbitMQ.Util;
 
 namespace RabbitMQ.Client.Impl
 {
+    // * DESCRIPTION TAKEN FROM MS REFERENCE SOURCE *
+    // https://github.com/microsoft/referencesource/blob/master/mscorlib/system/decimal.cs
+    // The lo, mid, hi, and flags fields contain the representation of the
+    // Decimal value. The lo, mid, and hi fields contain the 96-bit integer
+    // part of the Decimal. Bits 0-15 (the lower word) of the flags field are
+    // unused and must be zero; bits 16-23 contain must contain a value between
+    // 0 and 28, indicating the power of 10 to divide the 96-bit integer part
+    // by to produce the Decimal value; bits 24-30 are unused and must be zero;
+    // and finally bit 31 indicates the sign of the Decimal value, 0 meaning
+    // positive and 1 meaning negative.
+    readonly struct DecimalData
```

Comment:
could we move it inside WireFormatting as private class? It's not needed outside.

Reply:
Sure.